src/components/PopupWithForm.js

```diff
@@ -0,0 +1,52 @@
+import Popup from "./Popup";
+export default class PopupWithForm extends Popup {
+  constructor({ popupSelector, handleFormSubmit, loadingText }) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._inputList = this._popupElement.querySelectorAll(".modal__input");
+    this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popupForm.querySelector(
+      ".modal__button[type='submit']"
+    );
+    this._submitButtonText = this._submitButton.textContent;
+    this._loadingText = loadingText;
+  }
+
+  setInputValues(data) {
+    this._inputList.forEach((input) => {
+      input.value = data[input.name];
+    });
+  }
+
+  close() {
+    this._popupForm.reset();
+    super.close();
+  }
+  _getInputValues() {
+    this._newData = {};
+    this._inputList.forEach((inputElement) => {
+      this._newData[inputElement.name] = inputElement.value;
+    });
+    return this._newData;
+  }
+  setSubmitAction(action) {
```

Comment:
This class should not have such method

Reply:
deleting this, gives me an error in my index.js i use it on line 178 with confirmmodal, but if i take out setSubmitAction the whole function goes in error.

Reply:
You should use confirmation popup `PopupWithConfirmation` for handling the confirmation, not `PopupWithForm`